systemtest/src/main/java/io/strimzi/systemtest/resources/crd/kafkaclients/KafkaOauthExampleClients.java

```diff
@@ -8,52 +8,75 @@
+            this.oauthClientSecret = oauthClientSecret;
+            return self();
+        }
+
+        public Builder withOAuthTokenEndpointUri(String oauthTokenEndpointUri) {
+            this.oauthTokenEndpointUri = oauthTokenEndpointUri;
+            return self();
+        }
+
+        public Builder withUserName(String userName) {
+            this.userName = userName;
+            return self();
+        }
+
+        @Override
+        public KafkaOauthExampleClients build() {
+            return new KafkaOauthExampleClients(this);
+        }
+
+        @Override
+        protected KafkaOauthExampleClients.Builder self() {
+            return this;
+        }
     }
 
-    // from existing client create new client with random consumer group (immutability)
-    public KafkaOauthClientsResource(KafkaOauthClientsResource kafkaOauthClientsResource) {
-        super(kafkaOauthClientsResource);
-        this.oauthClientId = kafkaOauthClientsResource.oauthClientId;
-        this.oauthClientSecret = kafkaOauthClientsResource.oauthClientSecret;
-        this.oauthTokenEndpointUri = kafkaOauthClientsResource.oauthTokenEndpointUri;
-        this.userName = kafkaOauthClientsResource.userName;
-
+    private KafkaOauthExampleClients(KafkaOauthExampleClients.Builder builder) {
+        super(builder);
+        if (builder.oauthClientId == null || builder.oauthClientId.isEmpty()) throw new InvalidParameterException("OAuth client id is not set.");
+        if (builder.oauthClientSecret == null || builder.oauthClientSecret.isEmpty()) throw new InvalidParameterException("OAuth client secret is not set.");
+        if (builder.oauthTokenEndpointUri == null || builder.oauthTokenEndpointUri.isEmpty()) throw new InvalidParameterException("OAuth token endpoint url is not set.");
+        if (builder.userName == null || builder.userName.isEmpty()) builder.userName = builder.oauthClientId;
+
+        oauthClientId = builder.oauthClientId;
+        oauthClientSecret = builder.oauthClientSecret;
+        oauthTokenEndpointUri = builder.oauthTokenEndpointUri;
+        userName = builder.userName;
     }
 
-    // from existing client create new client with new specific consumer group and topicName (immutability)
-    public KafkaOauthClientsResource(KafkaOauthClientsResource kafkaOauthClientsResource, String topicName, String consumerGroup) {
-        super(kafkaOauthClientsResource, topicName, consumerGroup);
-        this.oauthClientId = kafkaOauthClientsResource.oauthClientId;
-        this.oauthClientSecret = kafkaOauthClientsResource.oauthClientSecret;
-        this.oauthTokenEndpointUri = kafkaOauthClientsResource.oauthTokenEndpointUri;
-        this.userName = kafkaOauthClientsResource.userName;
+    public void setTopicName(String topicName) {
+        this.topicName = topicName;
+    }
 
+    public void setConsumerGroup(String consumerGroup) {
+        this.consumerGroup = consumerGroup;
```

Comment:
I know that I have in my ExternalClients the setters, but that's not good practice. Maybe you could create sometimes that will do this
```
On the created instance of the client just modified the fields and create another.
KafkaOauthClientsResource client = new KafkaOauthClientsResourceBuilder(anotherClient)
    .editTopicName("something")
    .editConsumerName("something")
    .build();
```
Creating setters will again break the main meaning of builder pattern.